src/components/ListAritcles.jsx

```diff
@@ -1,29 +1,40 @@
 import { useState, useEffect} from "react";
 import { getArticles } from '../utils/api.js'
 
-const ListArticles = () => {
+const ListArticles = ({topicSelect}) => {
+    
     
     const [articles, setArticles] = useState([])
     const [isLoading, setIsLoading] = useState(true);
 
     useEffect(() => {
         setIsLoading(true);
-        getArticles(searchType).then((res) => {
+        getArticles().then((res) => {
             setArticles(res)
             setIsLoading(false);
         }).catch(() => {
             setIsLoading(false);
         })
-    }, [])
+    }, [topicSelect])
     
     if(isLoading) return <p>Loading...</p>
-    
+
+
+    let showAll = () => articles.map(article => (
+        <li className='listed-articles' key={article.article_id}>Article Name: {article.title}<br></br>Topic: {article.topic}</li>)
+    )
+
+    let filter = () => articles.filter(article => (article.topic === topicSelect)
```

Comment:
THIS IS NOT HOW YOU SHOULD BE DOING THIS. Imagine if you had 5 million posts and you had to filter through them every time you wanted to get a single topic!

You should be using `useParams` in this component...

`const { topic } = useParams()`